### PR TITLE
Fix handling of skipped binding constraints

### LINKED ISSUE #355: Traitement différent entre BC skipped et BC disabled
**Description**
Le fait qu'une contrainte couplante soit "skipped" (associées à des clusters tous désactivés par exemple) ne supprime pas totalement le test de vraisemblance de la contrainte sur la première semaine.
Ainsi, si une contrainte couplante est de type : "1.00 x cluster1 + 1.00 x cluster2 > b" et que b est strictement positif sur la première semaine alors le test de vraisemblance si cluster1 et cluster2 sont désactivés, dit qu'il est impossible que 0 + 0 > b et la simulation sort en erreur

**How to Reproduce**
Etude simple pour reproduire l'anomalie 
[test.zip](https://github.com/AntaresSimulatorTeam/Antares_Simulator/files/7014804/test.zip)

**Expected behavior**
Supprimer tout test de vraisemblance lorsqu'une contrainte couplante est de type "skipped" (la traiter comme si elle était "disabled").

**Screenshots and additional files**
If applicable, add screenshots to help explain your problem. You may also include a .zip file containing your study.

**OS and version**
 - OS: Windows
 - Antares Simulator version: 8.0.* branch

**Additional context**

### PR DESCRIPTION
**Pull request by flomnes**
Fix #355

### CODE REVIEW

File: src/libs/antares/study/constraint/constraint.hxx [hunk 1]
```diff
@@ -114,6 +114,11 @@ inline BindingConstraint::const_iterator BindingConstraint::end() const
     return pLinkWeights.end();
 }
 
+inline bool BindingConstraint::skipped() const
+{
+    return !(linkCount() > 0 || enabledClusterCount() > 0);
```
Comment (flomnes) [inline src/libs/antares/study/constraint/constraint.hxx]: `linkCount() == 0 && enabledClusterCount() == 0`

**Review by flomnes — COMMENTED** [review]
Add provided study to Antares_Simulator_Tests repository, run as end-to-end test.

File: src/ui/simulator/toolbox/components/datagrid/renderer/bindingconstraint/offsets.cpp [hunk 2]
```diff
@@ -158,9 +160,8 @@ IRenderer::CellStyle LinkOffsets::cellStyle(int x, int y) const
     case 2:
         return IRenderer::cellStyleConstraintWeightCount;
     case 3:
-        return ((study->uiinfo->constraint(x))->enabled()
-                && ((study->uiinfo->constraint(x))->linkCount() > 0
-                    || (study->uiinfo->constraint(x))->enabledClusterCount() > 0))
+        return (constraint->enabled()
+                && (constraint->linkCount() > 0 || constraint->enabledClusterCount() > 0))
```
Comment (flomnes) [inline src/ui/simulator/toolbox/components/datagrid/renderer/bindingconstraint/offsets.cpp]: use `constraint->skipped()`

File: src/libs/antares/study/constraint/constraint.hxx (hunk 1 above)
Comment (guilpier-code) [reply src/libs/antares/study/constraint/constraint.hxx]: I do agree with your proposal of change

File: src/ui/simulator/toolbox/components/datagrid/renderer/bindingconstraint/offsets.cpp (hunk 2 above)
Comment (guilpier-code) [reply src/ui/simulator/toolbox/components/datagrid/renderer/bindingconstraint/offsets.cpp]: I do agree with your proposal of change

File: src/libs/antares/study/constraint/constraint.hxx (hunk 1 above)
Comment (guilpier-code) [reply src/libs/antares/study/constraint/constraint.hxx]: This comment to tell gitHub I have requested changes (the same as those spotted by flomnes)

**Review by guilpier-code — CHANGES_REQUESTED** [review]

**Pull request merged**